### PR TITLE
283: Update NVCC docker to use official nvidia image

### LINKED ISSUE #283: Update NVCC docker to use official image

### PR DESCRIPTION
**Pull request by thearusable**
Close #283

### CODE REVIEW

File: ci/docker/ubuntu-20.04-nvidia-cpp.dockerfile [hunk 1]
```diff
@@ -1,9 +1,10 @@
 
+ARG compiler=11.0.3
```
Comment (cz4rs) [inline ci/docker/ubuntu-20.04-nvidia-cpp.dockerfile]: Why this version as default?

Comment (cz4rs) [reply ci/docker/ubuntu-20.04-nvidia-cpp.dockerfile]: Never mind, I can see this is consistent with `vt`.

**Review by cz4rs — APPROVED** [review]

File: ci/docker/ubuntu-20.04-nvidia-cpp.dockerfile (hunk 1 above)
Comment (thearusable) [reply ci/docker/ubuntu-20.04-nvidia-cpp.dockerfile]: It was the same in `vt`. I changed it to the version that is currently used on CI

**Review by cz4rs — APPROVED** [review]

**Review by JacobDomagala — APPROVED** [review]

File: .github/workflows/dockerimage-nvcc-11-ubuntu-mpich.yml [hunk 2]
```diff
@@ -49,7 +49,7 @@ jobs:
         message("::set-output name=timestamp::${current_date}")
     - uses: actions/cache@v1
       env:
-        cache-name: ubuntu-nvcc-11.2-cache
+        cache-name: ubuntu-11.2.0-cache
```
Comment (PhilMiller) [inline .github/workflows/dockerimage-nvcc-11-ubuntu-mpich.yml]: The cache name should probably specify that it's for nvcc

Comment (PhilMiller) [reply .github/workflows/dockerimage-nvcc-11-ubuntu-mpich.yml]: Other than this, I'm ready to Merge

Comment (thearusable) [reply .github/workflows/dockerimage-nvcc-11-ubuntu-mpich.yml]: @PhilMiller Ok, Fixed.

**Review by PhilMiller — APPROVED** [review]

**Pull request merged**